File: api/generate-claim.js
export default async function handler(req, res) {
  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method not allowed' })
  }

  try {
    const {
      county,
      property_address,
      surplus_amount,
      owner_name,
      claimant_name,
      claimant_address,
      claimant_email,
      claimant_phone,
      claimant_type,
      user_id
    } = req.body

    if (!user_id || !claimant_name || !claimant_address || !claimant_email) {
      return res.status(400).json({ error: 'Missing required fields' })
    }

    const SUPABASE_URL = process.env.SUPABASE_URL
    const SUPABASE_SERVICE_KEY = process.env.SUPABASE_SERVICE_KEY
    const PDFSHIFT_API_KEY = process.env.PDFSHIFT_API_KEY

    const today = new Date().toLocaleDateString('en-US', { year: 'numeric', month: 'long', day: 'numeric' })
    const formattedAmount = surplus_amount ? '$' + Number(surplus_amount).toLocaleString() : 'To Be Determined'

    // 1. Insert claim package record
    const insertRes = await fetch(`${SUPABASE_URL}/rest/v1/claim_packages`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'apikey': SUPABASE_SERVICE_KEY,
        'Authorization': `Bearer ${SUPABASE_SERVICE_KEY}`,
        'Prefer': 'return=representation'
      },
      body: JSON.stringify({
        user_id,
        county: county || '',
        property_address: property_address || '',
        claimant_name,
        claimant_type: claimant_type || 'Owner',
        surplus_amount: surplus_amount ? parseFloat(surplus_amount) : null,
        status: 'generating'
      })
    })

    const insertData = await insertRes.json()
    if (!insertRes.ok) throw new Error(JSON.stringify(insertData))
    const claimPackage = Array.isArray(insertData) ? insertData[0] : insertData

    // 2. Build HTML claim package from templates
    const html = `<!DOCTYPE html>
<html>
<head>
<meta charset="UTF-8">
<style>
  body { font-family: Georgia, serif; color: #1a1a1a; margin: 0; padding: 0; font-size: 12pt; line-height: 1.6; }
  .page { padding: 1in; min-height: 9in; page-break-after: always; }
  .page:last-child { page-break-after: avoid; }
  h1 { font-size: 16pt; text-align: center; text-transform: uppercase; letter-spacing: 0.1em; border-bottom: 2px solid #1a1a1a; padding-bottom: 0.5rem; margin-bottom: 1.5rem; }
  h2 { font-size: 13pt; text-transform: uppercase; letter-spacing: 0.05em; margin-top: 1.5rem; }
  .label { font-size: 10pt; text-transform: uppercase; letter-spacing: 0.08em; color: #555; margin-bottom: 0.2rem; }
  .value { font-size: 12pt; border-bottom: 1px solid #999; padding-bottom: 0.2rem; margin-bottom: 1rem; min-height: 1.2rem; }
  .signature-line { border-bottom: 1px solid #1a1a1a; margin-top: 2rem; margin-bottom: 0.25rem; width: 60%; }
  .two-col { display: grid; grid-template-columns: 1fr 1fr; gap: 1.5rem; }
  .header-info { text-align: right; margin-bottom: 2rem; }
  .county-address { margin-bottom: 2rem; }
  .re-line { margin: 1.5rem 0; font-weight: bold; }
  ol li { margin-bottom: 0.75rem; }
  .notary-box { border: 1px solid #1a1a1a; padding: 1rem; margin-top: 2rem; }
</style>
</head>
<body>

<!-- PAGE 1: COVER LETTER -->
<div class="page">
  <div class="header-info">
    <div>${claimant_name}</div>
    <div>${claimant_address}</div>
    <div>${claimant_email}</div>
    <div>${claimant_phone || ''}</div>
    <div style="margin-top:0.5rem">${today}</div>
  </div>

  <div class="county-address">
    <div>${county} County Clerk / Treasurer</div>
    <div>${county} County Courthouse</div>
    <div>${county}, Georgia</div>
  </div>

  <div class="re-line">RE: Claim for Surplus Funds — ${property_address || 'Property on File'} — Original Owner: ${owner_name || 'On File'}</div>

  <p>Dear ${county} County Clerk/Treasurer,</p>

  <p>I am writing to formally request the release of surplus funds in the amount of <strong>${formattedAmount}</strong> resulting from the tax sale or foreclosure of the above-referenced property. I am the ${claimant_type} and am entitled to claim these funds pursuant to applicable Georgia statutes.</p>

  <p>Enclosed with this letter you will find the following documents in support of this claim:</p>
  <ol>
    <li>Completed Surplus Funds Claim Form</li>
    <li>Proof of Ownership / Entitlement Affidavit</li>
    <li>Valid government-issued identification (to be attached)</li>
    <li>Supporting documentation establishing claimant's right to funds (to be attached)</li>
  </ol>

  <p>Please review the enclosed materials and process this claim at your earliest convenience. Should you require any additional information or documentation, please do not hesitate to contact me at ${claimant_email} or ${claimant_phone || 'the contact information above'}.</p>

  <p>Thank you for your time and attention to this matter.</p>

  <p style="margin-top: 2rem">Respectfully submitted,</p>
  <div class="signature-line"></div>
  <div>${claimant_name}</div>
  <div style="font-size:10pt; color:#555">${claimant_type} / Claimant</div>
  <div style="font-size:10pt; color:#555">${today}</div>
</div>

<!-- PAGE 2: CLAIM FORM -->
<div class="page">
  <h1>Surplus Funds Claim Form</h1>

  <h2>Section 1 — Claimant Information</h2>
  <div class="label">Full Legal Name</div>
  <div class="value">${claimant_name}</div>
  <div class="label">Mailing Address</div>
  <div class="value">${claimant_address}</div>
  <div class="two-col">
    <div>
      <div class="label">Email Address</div>
      <div class="value">${claimant_email}</div>
    </div>
    <div>
      <div class="label">Phone Number</div>
      <div class="value">${claimant_phone || ''}</div>
    </div>
  </div>
  <div class="label">Relationship to Property Owner</div>
  <div class="value">${claimant_type}</div>

  <h2>Section 2 — Property Information</h2>
  <div class="label">Original Property Owner</div>
  <div class="value">${owner_name || ''}</div>
  <div class="label">Property Address</div>
  <div class="value">${property_address || ''}</div>
  <div class="two-col">
    <div>
      <div class="label">County</div>
      <div class="value">${county}</div>
    </div>
    <div>
      <div class="label">Surplus Amount Claimed</div>
      <div class="value">${formattedAmount}</div>
    </div>
  </div>

  <h2>Section 3 — Certification</h2>
  <p>I, the undersigned, hereby certify under penalty of perjury that all information provided in this claim form is true, accurate, and complete to the best of my knowledge. I am legally entitled to claim the surplus funds described herein.</p>

  <div class="two-col" style="margin-top:2rem">
    <div>
      <div class="signature-line"></div>
      <div style="font-size:10pt">Claimant Signature</div>
    </div>
    <div>
      <div class="signature-line"></div>
      <div style="font-size:10pt">Date</div>
    </div>
  </div>
</div>

<!-- PAGE 3: AFFIDAVIT -->
<div class="page">
  <h1>Proof of Ownership / Entitlement Affidavit</h1>

  <p>STATE OF GEORGIA<br>COUNTY OF ${county}</p>

  <p>PERSONALLY APPEARED before me, the undersigned notary public, <strong>${claimant_name}</strong>, who, being duly sworn, deposes and states as follows:</p>

  <ol>
    <li>My name is <strong>${claimant_name}</strong> and I reside at <strong>${claimant_address}</strong>.</li>
    <li>I am the <strong>${claimant_type}</strong> of the property formerly owned by <strong>${owner_name || '________________'}</strong>, located at <strong>${property_address || '________________'}</strong>, in ${county} County, Georgia.</li>
    <li>Surplus funds in the amount of <strong>${formattedAmount}</strong> were generated as a result of a tax sale or foreclosure proceeding related to the above-referenced property.</li>
    <li>I am legally entitled to claim and receive these surplus funds pursuant to applicable Georgia law.</li>
    <li>No other person or entity has a superior claim to these surplus funds to the best of my knowledge.</li>
    <li>All information provided in support of this claim is true, accurate, and complete.</li>
  </ol>

  <p>FURTHER AFFIANT SAYETH NOT.</p>

  <div style="margin-top: 2rem">
    <div class="signature-line"></div>
    <div>${claimant_name} — Affiant</div>
  </div>

  <div class="notary-box">
    <p><strong>NOTARY PUBLIC</strong></p>
    <p>Sworn to and subscribed before me this ______ day of ________________, 20____.</p>
    <div class="signature-line"></div>
    <div style="font-size:10pt">Notary Public Signature</div>
    <div style="margin-top: 1rem; font-size:10pt">My Commission Expires: ______________________</div>
    <div style="margin-top: 0.5rem; font-size:10pt">[NOTARY SEAL]</div>
  </div>
</div>

<!-- PAGE 4: FILING INSTRUCTIONS -->
<div class="page">
  <h1>Filing Instructions</h1>
  <h2>${county} County Surplus Funds Claim — Step by Step Guide</h2>

  <p>Follow these steps carefully to file your surplus funds claim with ${county} County:</p>

  <ol>
    <li><strong>Gather your documents.</strong> Make sure you have all enclosed documents completed and signed. You will also need a valid government-issued photo ID (driver's license or passport) and any documents proving your relationship to the property (deed, probate records, etc.).</li>

    <li><strong>Get the Affidavit notarized.</strong> Take the Proof of Ownership / Entitlement Affidavit to a notary public. Do NOT sign it until you are in front of the notary. Notary services are available at most banks, UPS stores, and county courthouses.</li>

    <li><strong>Make copies.</strong> Make copies of all documents before submitting. Keep the originals for your records and submit copies, unless the county specifically requires originals.</li>

    <li><strong>Submit your claim.</strong> Deliver your completed claim package to the ${county} County Clerk or Treasurer's office. You may submit in person, by certified mail, or as directed by the county. Request a receipt or confirmation of filing.</li>

    <li><strong>Follow up.</strong> Processing times vary by county, typically 30–90 days. Follow up with the county office if you have not received a response within 60 days.</li>

    <li><strong>Receive your funds.</strong> Once approved, funds are typically issued by check to the claimant's address on file. Make sure your mailing address is current and accurate.</li>
  </ol>

  <h2>Important Notes</h2>
  <ul>
    <li>Deadlines apply — surplus funds may be forfeited if not claimed within the statutory period. File promptly.</li>
    <li>If your claim is denied, you have the right to appeal. Contact a licensed attorney for assistance.</li>
    <li>This package was prepared by SurplusRecoveryPro. For support, contact us at support@surplusrecoverypro.site.</li>
  </ul>

  <p style="margin-top: 2rem; font-size: 10pt; color: #555; border-top: 1px solid #ccc; padding-top: 1rem;">
    Generated by SurplusRecoveryPro on ${today} for ${claimant_name}. This document is for informational purposes and does not constitute legal advice. Consult a licensed attorney for legal guidance specific to your situation.
  </p>
</div>

</body>
</html>`

    // 3. Convert HTML to PDF using PDFShift
    const pdfRes = await fetch('https://api.pdfshift.io/v3/convert/pdf', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': 'Basic ' + Buffer.from(`api:${PDFSHIFT_API_KEY}`).toString('base64')
      },
      body: JSON.stringify({
        source: html,
        landscape: false,
        use_print: false,
        margin: { top: '0.25in', bottom: '0.25in', left: '0.25in', right: '0.25in' }
      })
    })

    if (!pdfRes.ok) {
      const pdfError = await pdfRes.text()
      throw new Error('PDFShift error: ' + pdfError)
    }

    const pdfBuffer = await pdfRes.arrayBuffer()

    // 4. Upload PDF to Supabase Storage
    const fileName = `claim-${claimPackage.id}-${Date.now()}.pdf`
    const uploadRes = await fetch(`${SUPABASE_URL}/storage/v1/object/claim-packages/${fileName}`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/pdf',
        'apikey': SUPABASE_SERVICE_KEY,
        'Authorization': `Bearer ${SUPABASE_SERVICE_KEY}`,
        'x-upsert': 'true'
      },
      body: pdfBuffer
    })

    if (!uploadRes.ok) {
      const uploadError = await uploadRes.text()
      throw new Error('Storage upload error: ' + uploadError)
    }

    // 5. Build public URL and update record
    const pdfUrl = `${SUPABASE_URL}/storage/v1/object/public/claim-packages/${fileName}`

    await fetch(`${SUPABASE_URL}/rest/v1/claim_packages?id=eq.${claimPackage.id}`, {
      method: 'PATCH',
      headers: {
        'Content-Type': 'application/json',
        'apikey': SUPABASE_SERVICE_KEY,
        'Authorization': `Bearer ${SUPABASE_SERVICE_KEY}`
      },
      body: JSON.stringify({ status: 'complete', pdf_url: pdfUrl })
    })

    return res.status(200).json({
      success: true,
      package_id: claimPackage.id,
      message: 'Claim package generated successfully'
    })

  } catch (err) {
    console.error('Claim generation error:', err)
    return res.status(500).json({ error: 'Failed to generate claim package: ' + err.message })
  }
}

File: agent/generate_claim.py
"""
SurplusRecoveryPro - Claim Package Generator
Generates a complete PDF claim package for a surplus funds case.
"""

from reportlab.lib.pagesizes import letter
from reportlab.lib.styles import getSampleStyleSheet, ParagraphStyle
from reportlab.lib.units import inch
from reportlab.lib import colors
from reportlab.platypus import SimpleDocTemplate, Paragraph, Spacer, PageBreak, HRFlowable
from reportlab.lib.enums import TA_CENTER, TA_LEFT, TA_JUSTIFY
from datetime import datetime
import os
import uuid


def generate_claim_package(case_data, claimant_data, output_path=None):
    if not output_path:
        package_id = str(uuid.uuid4())[:8].upper()
        output_path = f"/tmp/claim_package_{package_id}.pdf"

    doc = SimpleDocTemplate(
        output_path,
        pagesize=letter,
        rightMargin=inch,
        leftMargin=inch,
        topMargin=inch,
        bottomMargin=inch
    )

    styles = getSampleStyleSheet()

    title_style = ParagraphStyle('CustomTitle', parent=styles['Title'], fontSize=16, fontName='Helvetica-Bold', spaceAfter=12, alignment=TA_CENTER)
    heading_style = ParagraphStyle('CustomHeading', parent=styles['Heading1'], fontSize=13, fontName='Helvetica-Bold', spaceBefore=16, spaceAfter=8, textColor=colors.HexColor('#1a365d'))
    subheading_style = ParagraphStyle('CustomSubHeading', parent=styles['Heading2'], fontSize=11, fontName='Helvetica-Bold', spaceBefore=10, spaceAfter=6)
    normal_style = ParagraphStyle('CustomNormal', parent=styles['Normal'], fontSize=10, fontName='Helvetica', spaceAfter=6, leading=14)
    body_style = ParagraphStyle('CustomBody', parent=styles['Normal'], fontSize=10, fontName='Helvetica', spaceAfter=8, leading=16, alignment=TA_JUSTIFY)
    center_style = ParagraphStyle('Center', parent=styles['Normal'], fontSize=10, fontName='Helvetica', alignment=TA_CENTER, spaceAfter=6)

    today = datetime.now().strftime("%B %d, %Y")
    county = case_data.get('county', 'Unknown')
    property_address = case_data.get('property_address', 'Unknown')
    owner_name = case_data.get('owner_name', 'Unknown')
    surplus_amount = case_data.get('surplus_amount', 0)
    parcel_id = case_data.get('parcel_id', 'N/A')
    sale_date = case_data.get('sale_date', 'N/A')
    case_number = case_data.get('case_number', 'N/A')
    claimant_name = claimant_data.get('claimant_name', '')
    claimant_address = claimant_data.get('claimant_address', '')
    claimant_email = claimant_data.get('claimant_email', '')
    claimant_phone = claimant_data.get('claimant_phone', '')
    claimant_type = claimant_data.get('claimant_type', 'owner').title()

    story = []

    # PAGE 1: COVER LETTER
    story.append(Paragraph("SURPLUS RECOVERY CLAIM PACKAGE", title_style))
    story.append(Paragraph(f"{county} County, Georgia", center_style))
    story.append(HRFlowable(width="100%", thickness=2, color=colors.HexColor('#1a365d')))
    story.append(Spacer(1, 20))
    story.append(Paragraph(claimant_name, normal_style))
    story.append(Paragraph(claimant_address, normal_style))
    story.append(Paragraph(claimant_email, normal_style))
    story.append(Paragraph(claimant_phone, normal_style))
    story.append(Spacer(1, 10))
    story.append(Paragraph(today, normal_style))
    story.append(Spacer(1, 10))
    story.append(Paragraph(f"{county} County Tax Commissioner", normal_style))
    story.append(Paragraph(f"{county} County, Georgia", normal_style))
    story.append(Spacer(1, 16))
    story.append(Paragraph(f"<b>RE: Claim for Surplus Funds — {property_address}</b>", normal_style))
    story.append(Paragraph(f"<b>Parcel ID:</b> {parcel_id} | <b>Sale Date:</b> {sale_date} | <b>Estimated Surplus:</b> ${surplus_amount:,.2f}", normal_style))
    story.append(Spacer(1, 16))
    story.append(Paragraph("Dear Tax Commissioner,", normal_style))
    story.append(Spacer(1, 8))
    story.append(Paragraph(f"I, {claimant_name}, am writing to formally claim the surplus funds generated from the tax sale of the above-referenced property located at {property_address}, {county} County, Georgia.", body_style))
    story.append(Paragraph(f"I am submitting this claim as the <b>{claimant_type}</b> of the above property. Enclosed with this letter you will find the required supporting documentation including an Affidavit of Claim and Motion for Disbursement of Funds.", body_style))
    story.append(Paragraph(f"I respectfully request that the surplus funds in the amount of <b>${surplus_amount:,.2f}</b> be disbursed to me at the address listed above. Please contact me at {claimant_email} or {claimant_phone} if you require any additional documentation.", body_style))
    story.append(Spacer(1, 16))
    story.append(Paragraph("Respectfully submitted,", normal_style))
    story.append(Spacer(1, 40))
    story.append(Paragraph("_______________________________", normal_style))
    story.append(Paragraph(claimant_name, normal_style))
    story.append(Paragraph(f"Claimant ({claimant_type})", normal_style))
    story.append(PageBreak())

    # PAGE 2: AFFIDAVIT
    story.append(Paragraph("AFFIDAVIT OF CLAIM FOR SURPLUS FUNDS", title_style))
    story.append(Paragraph(f"State of Georgia, {county} County", center_style))
    story.append(HRFlowable(width="100%", thickness=2, color=colors.HexColor('#1a365d')))
    story.append(Spacer(1, 20))
    story.append(Paragraph(f"I, {claimant_name}, being duly sworn, depose and state as follows:", body_style))
    for item in [
        f"1. My name is <b>{claimant_name}</b> and I reside at <b>{claimant_address}</b>.",
        f"2. I am the <b>{claimant_type}</b> of the property located at <b>{property_address}</b>, {county} County, Georgia, Parcel ID: {parcel_id}.",
        f"3. The above property was sold at a tax sale on <b>{sale_date}</b>, generating surplus funds in the estimated amount of <b>${surplus_amount:,.2f}</b>.",
        f"4. I am entitled to claim these surplus funds as the {claimant_type} of record.",
        "5. I have not transferred, assigned, or conveyed my rights to these surplus funds to any other party.",
        "6. To the best of my knowledge, there are no other parties with a superior claim to these surplus funds.",
        "7. All statements made in this affidavit are true and correct to the best of my knowledge and belief."
    ]:
        story.append(Paragraph(item, body_style))
    story.append(Spacer(1, 20))
    story.append(Paragraph("FURTHER AFFIANT SAYETH NOT.", normal_style))
    story.append(Spacer(1, 30))
    story.append(Paragraph("_______________________________", normal_style))
    story.append(Paragraph(f"{claimant_name}, Affiant", normal_style))
    story.append(Spacer(1, 30))
    story.append(Paragraph("<b>Sworn to and subscribed before me this _____ day of _____________, 20____</b>", normal_style))
    story.append(Spacer(1, 30))
    story.append(Paragraph("_______________________________", normal_style))
    story.append(Paragraph("Notary Public", normal_style))
    story.append(Paragraph("My Commission Expires: _______________", normal_style))
    story.append(PageBreak())

    # PAGE 3: MOTION FOR DISBURSEMENT
    story.append(Paragraph("MOTION FOR DISBURSEMENT OF SURPLUS FUNDS", title_style))
    story.append(Paragraph(f"In the Superior Court of {county} County, State of Georgia", center_style))
    story.append(HRFlowable(width="100%", thickness=2, color=colors.HexColor('#1a365d')))
    story.append(Spacer(1, 20))
    story.append(Paragraph(f"<b>Case/Parcel:</b> {parcel_id}", normal_style))
    story.append(Paragraph(f"<b>Property:</b> {property_address}", normal_style))
    story.append(Paragraph(f"<b>Sale Date:</b> {sale_date}", normal_style))
    story.append(Paragraph(f"<b>Surplus Amount:</b> ${surplus_amount:,.2f}", normal_style))
    story.append(Spacer(1, 16))
    story.append(Paragraph(f"NOW COMES {claimant_name.upper()}, Movant, and respectfully moves this Court to order the disbursement of surplus funds held by {county} County, Georgia, and in support thereof states:", body_style))
    for item in [
        f"1. On or about <b>{sale_date}</b>, the property located at <b>{property_address}</b>, {county} County, Georgia was sold at a tax sale.",
        f"2. The sale generated surplus funds in the amount of approximately <b>${surplus_amount:,.2f}</b> after satisfaction of all outstanding tax obligations.",
        f"3. Movant, <b>{claimant_name}</b>, is the {claimant_type} of the above-referenced property and is entitled to receive the surplus funds.",
        "4. Movant has provided all required documentation to support this claim, including an Affidavit of Claim and proof of identity.",
        "5. No other party has filed a claim to these surplus funds during the applicable claims period, or Movant's claim is superior to all other claims.",
    ]:
        story.append(Paragraph(item, body_style))
    story.append(Spacer(1, 10))
    story.append(Paragraph(f"WHEREFORE, Movant respectfully requests that this Court enter an Order directing {county} County to disburse the surplus funds in the amount of ${surplus_amount:,.2f} to {claimant_name} at {claimant_address}.", body_style))
    story.append(Spacer(1, 30))
    story.append(Paragraph("Respectfully submitted,", normal_style))
    story.append(Spacer(1, 40))
    story.append(Paragraph("_______________________________", normal_style))
    story.append(Paragraph(claimant_name, normal_style))
    story.append(Paragraph(claimant_address, normal_style))
    story.append(Paragraph(claimant_email, normal_style))
    story.append(Paragraph(f"Date: {today}", normal_style))
    story.append(PageBreak())

    # PAGE 4: FILING INSTRUCTIONS
    story.append(Paragraph("FILING INSTRUCTIONS", title_style))
    story.append(Paragraph(f"{county} County, Georgia", center_style))
    story.append(HRFlowable(width="100%", thickness=2, color=colors.HexColor('#1a365d')))
    story.append(Spacer(1, 20))
    story.append(Paragraph("Follow these steps to file your surplus funds claim:", heading_style))
    for title, content in [
        ("<b>Step 1 — Notarize Your Affidavit</b>", "Take the Affidavit of Claim (Page 2) to a notary public. Sign in front of the notary and have them complete the notarization section. Banks, UPS Stores, and many courthouses offer notary services."),
        ("<b>Step 2 — Gather Supporting Documents</b>", "Collect: Government-issued photo ID, Proof of ownership or relationship to property (deed, probate documents), Social Security Number or Tax ID for disbursement."),
        ("<b>Step 3 — Submit Your Claim Package</b>", f"Deliver this complete package to the {county} County Tax Commissioner's office in person or by certified mail. Keep copies of everything."),
        ("<b>Step 4 — Follow Up</b>", "Follow up with the Tax Commissioner's office within 2-3 weeks to confirm receipt. Processing typically takes 30-90 days."),
        ("<b>Step 5 — Receive Payment</b>", "Upon approval, the county will issue a check or direct deposit for the surplus funds amount."),
    ]:
        story.append(Paragraph(title, subheading_style))
        story.append(Paragraph(content, body_style))

    story.append(Spacer(1, 16))
    story.append(HRFlowable(width="100%", thickness=1, color=colors.grey))
    story.append(Spacer(1, 10))
    story.append(Paragraph("<b>Important Notes:</b>", subheading_style))
    for note in [
        "- Claim deadlines vary by county. File as soon as possible.",
        "- If you have an attorney, have them review this package before filing.",
        "- Keep certified copies of all documents you submit.",
        "- Always verify county-specific requirements with the Tax Commissioner's office.",
    ]:
        story.append(Paragraph(note, normal_style))

    story.append(Spacer(1, 20))
    story.append(Paragraph(f"Package generated by SurplusRecoveryPro on {today}", center_style))

    doc.build(story)
    return output_path


if __name__ == "__main__":
    case_data = {
        "county": "Fulton",
        "property_address": "123 Peachtree St NW, Atlanta, GA 30303",
        "owner_name": "John Smith",
        "surplus_amount": 15420.50,
        "parcel_id": "14-0075-0002-019-3",
        "sale_date": "January 15, 2024",
        "case_number": "FC-2024-001234"
    }
    claimant_data = {
        "claimant_name": "John Smith",
        "claimant_address": "456 Oak Avenue, Atlanta, GA 30301",
        "claimant_email": "john.smith@email.com",
        "claimant_phone": "(404) 555-0123",
        "claimant_type": "owner"
    }
    output = generate_claim_package(case_data, claimant_data)
    print(f"Claim package generated: {output}")
